### PR TITLE
Typed trainer config

### PR DESCRIPTION
**Pull request by nishu-builder**
Introduces pydantic types for trainer configs and makes use of them in train.py / trainer.py. If we like this approach, I'm happy to expand it to the overall config passed to train.py (train_job)

Attempts to follow the config-typing precedent set here [here](https://github.com/Metta-AI/metta/commit/1e52c68a45a2c03337927df955d6bed0e5b8b480), and does some very minor refactoring with it

It also introduces some validation of the runs sweep configs create

The tests cover:
- Config conversion / compatibility: DictConfig <-> Pydantic
- Hydra's behaviors with config inheritance are reflected in the parsed Pydantic objects
- New constraints: fields must be of the right type, must adhere to new validation rules, extra fields are disallowed
- Parsing all train configs in config/trainer/

On defaults:
Previously, in a few places in trainer.py, we assigned default values for some parameters if they were missing from trainer configs (`prio_beta`, `vf_clip_coef`, etc). Alex expressed an opinion that substantive values (e.g. learning rate constant) should exist in configs, not code.

Richard recently consolidated all our trainer configs into a single one [here](https://github.com/Metta-AI/metta/commit/6c4b5e22298c8aded6e9b89ea8c670db348b5dd4). He supplied values for nearly all the params that we previously specified in-code defaults for. I've added values to his config for the few remaining ones.

[Asana Task](https://app.asana.com/1/1209016784099267/project/1210348820405981/task/1210644465442528)

### CODE REVIEW

File: metta/rl/trainer_config.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    gae_lambda: float = Field(ge=0, le=1.0)
+    gamma: float = Field(ge=0, le=1.0)
+
+    # Optimizer and scheduler
+    optimizer: OptimizerConfig
+    lr_scheduler: LRSchedulerConfig
+
+    # Training parameters
+    max_grad_norm: float = Field(gt=0)
+    vf_clip_coef: float = Field(ge=0)
```
[the remark this answers is not in the document]
Comment (relh) [reply metta/rl/trainer_config.py:77]: Yeah! Happy to do that. It should be a simple find+replace in the various parts of the trainer code.

My hope is that this leads to there being no TrainerConfig eventually and that most pieces have other homes

[the remark this answers is not in the document]
Comment (nishu-builder) [reply metta/rl/trainer_config.py:77]: i'll submit another PR for these if that's alright

[the remark this answers is not in the document]
Comment (nishu-builder) [reply metta/rl/trainer_config.py:77]: https://github.com/Metta-AI/metta/pull/1193/files

File: metta/rl/trainer.py [hunk 2]
```diff
@@ -1079,7 +1082,7 @@ def _make_experience_buffer(self):
 
         # Calculate minibatch parameters
         minibatch_size = trainer_cfg.minibatch_size
-        max_minibatch_size = trainer_cfg.get("max_minibatch_size", minibatch_size)
+        max_minibatch_size = getattr(trainer_cfg, "max_minibatch_size", minibatch_size)
```
[the remark this answers is not in the document]
Comment (Barraketh) [reply metta/rl/trainer.py]: @nishu-builder - this seems like an issue - can you verify that this works?
